src/main/java/de/tum/in/www1/artemis/service/metis/PostingService.java

```diff
@@ -229,4 +233,57 @@ else if (authorizationCheckService.isTeachingAssistantInCourse(postingCourse, po
     }
 
     protected abstract String getEntityName();
+
+    /**
+     * Gets the list of logins for users mentioned in a posting.
+     * Throws an exception, if a mentioned user is not part of the course.
+     *
+     * @param course         course of the posting
+     * @param postingContent content of the posting
+     * @return the list of logins of mentioned users
+     */
+    protected List<String> parseUserMentions(@NotNull Course course, String postingContent) {
+        // Define a regular expression to match text enclosed in [user]...[/user] tags, along with login inside parentheses () within those tags.
+        // It makes use of the possessive quantifier "*+" to avoid backtracking and increase performance.
+        // Explanation:
+        // - "\\[user\\]" matches the literal string "[user]".
+        // - "([^\\[\\]()]*+)" captures any characters that are not '[', ']', '(', or ')' zero or more times. This captures the full name the user mention.
+        // - "\\(?" matches the literal '(' character.
+        // - "([^\\[\\]()]*+)" captures any characters that are not '[', ']', '(', or ')' zero or more times. This captures the content within parentheses.
+        // - "\\)?" matches the literal ')' character.
+        // - "\\[/user\\]" matches the literal string "[/user]".
+        String regex = "\\[user\\]([^\\[\\]()]*+)\\(?([^\\[\\]()]*+)\\)?\\[/user\\]";
+
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(Optional.ofNullable(postingContent).orElse(""));
+
+        Map<String, String> matches = new HashMap<>();
+
+        // Find and save all matches in the list
+        while (matcher.find()) {
+            String fullName = matcher.group(1);
+            String userLogin = matcher.group(2);
+
+            matches.put(userLogin, fullName);
+        }
+
+        List<User> mentionedUsers = userRepository.findAllByLogins(matches.keySet());
+
+        if (mentionedUsers.size() != matches.size()) {
+            throw new BadRequestAlertException("At least one of the mentioned users does not exist", METIS_POST_ENTITY_NAME, "invalidUserMention");
+        }
+
+        mentionedUsers.forEach(user -> {
+            if (!user.getName().equals(matches.get(user.getLogin()))) {
+                throw new BadRequestAlertException("The name " + matches.get(user.getLogin()) + " does not match the user's full name", METIS_POST_ENTITY_NAME,
```

Comment:
Can you include the actual name in the error message?

Reply:
I added it